src/guides/gelato-smart-contract-automation/index.md

```diff
@@ -0,0 +1,449 @@
+
+## Download VS Code
+
+Feel free to use whatever IDE you want, but we highly recommend using VS Code! You can run through most of this tutorial using the Truffle extension to create, build, and deploy your smart contracts, all without using the CLI! You can read more about it [here](https://trufflesuite.com/blog/build-on-web3-with-truffle-vs-code-extension/).
+
+## Get Some Test Eth
+
+In order to deploy to the public testnets, you'll need some test Eth to cover your gas fees! Make sure you're getting test Eth for either [Goerli](https://goerli-faucet.pk910.de/) or Sepolia.
+
+## Set Up Your Project
+
+Since we're building on our NFT rental marketplace from episode 4, start by cloning the base repository.
+
+```shell
+git clone https://github.com/trufflesuite/unleashed_nft_rental_marketplace.git gelato-demo
+```
+
+We'll only be working with contracts, so hop into the `truffle` directory and install our package dependencies.
+
+```shell
+cd gelato-demo/truffle
+npm i
+```
+
+## Write the automated task
+
+In our marketplace, we will remove listings that are no longer available. A listing becomes unavailable when the end date of the listing has passed. Add this method to `Marketplace.sol`: 
+
+```javascript
+function removeUnavailableListings(Listing[] memory unavailableListings) public {
+    for (uint i = 0; i < unavailableListings.length; i++) {
+        unlistNFT(unavailableListings[i].nftContract, unavailableListings[i].tokenId);
+    }
+}
+```
+
+It steps through a list of unavailable listings and calls our already written `unlistNFT` function, which will remove the NFT from our marketplace's list of listings and decrement the total number of listings available.
+
+## Write `IResolver`
+
+Because our automated task takes in the dynamic argument `unavailableListings` and is dependent on on a few conditions, we'll need to write a resolver. The first thing we will do is create a resolver interface. This isn't explicitly necessary, but is good practice if you have to write multiple resolvers for different smart contracts.
+
+```shell
+truffle create contract IResolver
+```
+Then, paste in the following code:
+
+```javascript
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.4.22 <0.9.0;
+
+interface IResolver {
+    function checker()
+        external
+        view
+        returns (bool canExec, bytes memory execPayload);
+}
+```
+
+And have `Marketplace.sol` extend that contract:
+
+```javascript
+import "./IResolver.sol";
+
+contract Marketplace is ReentrancyGuard, IResolver {
+```
+
+## Write the resolver
+
+Now, we can get to actually writing our resolver! It should return two things:
+
+1. bool `canExec` : whether Gelato should execute the task.
+2. bytes `execPayload` :  data that executors should use for the execution.
+
+Add this method:
+
+```javascript
+function checker()
+    external
+    view 
+    override returns (bool canExec, bytes memory execPayload) {
```

Comment:
Low-ish priority but improving the indention will help the readability here (solidity's docs have a good [style guide](https://docs.soliditylang.org/en/v0.8.17/style-guide.html#function-declaration) for handling multiline output params, etc)